hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/WatermarkSourceUtil.java

```diff
@@ -33,59 +33,69 @@
- * downstream will exclude this processor from watermark coalescing.
+ *
+ * It can happen that some partition does not have any events at all while
+ * others do or the processor doesn't get any external partitions assigned
+ * to it. If we simply wait for the timestamps in all partitions to advance
+ * to some point, we won't be emitting any watermarks. This utility
+ * supports the <em>idle timeout</em>: if there's no new data from a
+ * partition after the timeout elapses, it will be marked as <em>idle</em>,
+ * allowing the processor's watermark to advance as if that partition didn't
+ * exist. If all partitions are idle or there are no partitions, the
+ * processor will emit a special <em>idle message</em> and the downstream
+ * will exclude this processor from watermark coalescing.
  *
  * <h3>Usage</h3>
- * API is designed to be used as a flat-mapping step in {@link Traverser}. Your
- * source might follow this pattern:
+ *
+ * The API is designed to be used as a flat-mapping step in the {@link
+ * Traverser} that holds the output data. Your source can follow this
+ * pattern:
  *
  * <pre>{@code
- *   public boolean complete() {
- *       if (traverser == null) {
- *           List<Record> records = poll(); // get a batch of events from external source
- *           if (records.isEmpty()) {
- *               traverser = watermarkSourceUtil.handleNoEvent();
- *           } else {
- *               traverser = traverserIterable(records)
- *                   .flatMap(event -> watermarkSourceUtil.handleEvent(event, event.getPartition()));
- *           }
- *           traverser = traverser.onFirstNull(() -> traverser = null);
- *       }
- *       emitFromTraverser(traverser, event -> {
- *           if (!(event instanceof Watermark)) {
- *               // store your offset after event was emitted
- *               offsetsMap.put(event.getPartition(), event.getOffset());
- *           }
- *       });
- *       return false;
- *   }
+ * public boolean complete() {
+ *     if (traverser == null) {
+ *         List<Record> records = poll();
+ *         if (records.isEmpty()) {
+ *             traverser = wmSourceUtil.handleNoEvent();
+ *         } else {
+ *             traverser = traverserIterable(records)
+ *                 .flatMap(event -> wmSourceUtil.handleEvent(
+ *                      event, event.getPartition()));
+ *         }
+ *         traverser = traverser.onFirstNull(() -> traverser = null);
+ *     }
+ *     emitFromTraverser(traverser, event -> {
+ *         if (!(event instanceof Watermark)) {
+ *             // store your offset after event was emitted
+ *             offsetsMap.put(event.getPartition(), event.getOffset());
+ *         }
+ *     });
+ *     return false;
+ * }
  * }</pre>
  *
  * Other methods:
- * <ul>
- *     <li>Call {@link #increasePartitionCount} to set your partition count
+ * <ul><li>
+ *     Call {@link #increasePartitionCount} to set your partition count
```

Comment:
The `<li>`'s at the begginings of the lines resemble a bullet. Now it looks more like two independent paragraphs...